cpp/simulations/abm_braunschweig.cpp

```diff
@@ -0,0 +1,267 @@
+{
+    std::vector<std::string> strings;
+    boost::replace_all(string, ";;", ";-1;");
+    boost::split(strings, string, boost::is_any_of(";"));
+    std::transform(strings.begin(), strings.end(), std::back_inserter(*row), [&](std::string s) {
+        return std::stoi(s);
+    });
+}
+
+mio::abm::LocationType get_location_type(uint32_t acitivity_end)
+{
+    mio::abm::LocationType type;
+    switch (acitivity_end) {
+    case 1:
+        type = mio::abm::LocationType::Work;
+        break;
+    case 2:
+        type = mio::abm::LocationType::School;
+        break;
+    case 3:
+        type = mio::abm::LocationType::BasicsShop;
+        break;
+    case 4:
+        type = mio::abm::LocationType::SocialEvent; // Freizeit
+        break;
+    case 5:
+        type = mio::abm::LocationType::Home; // Private Erledigung
+        break;
+    case 6:
+        type = mio::abm::LocationType::SocialEvent; // Sonstiges
+        break;
+    default:
+        type = mio::abm::LocationType::Home;
+        break;
+    }
+    return type;
+}
+
+void create_world_from_data(mio::abm::World& world, const std::string& filename)
+{
+    // File pointer
+    std::fstream fin;
+
+    // Open an existing file
+    fin.open(filename, std::ios::in);
+    std::vector<int32_t> row;
+    std::vector<std::string> row_string;
+    std::string line;
+
+    std::getline(fin, line);
+    std::vector<std::string> titles;
+    boost::split(titles, line, boost::is_any_of(";"));
+    uint32_t count                        = 0;
+    std::map<std::string, uint32_t> index = {};
+    for (auto const& title : titles) {
+        index.insert({title, count});
+        row_string.push_back(title);
+        count++;
+    }
+    std::map<uint32_t, mio::abm::LocationId> locations = {};
+    std::map<uint32_t, mio::abm::Person&> persons      = {};
+
+    while (std::getline(fin, line)) {
+        row.clear();
+
+        // read columns in this row
+        split_line(line, &row);
+
+        uint32_t person_id   = row[index["puid"]];
+        uint32_t age         = row[index["age"]]; // TODO
+        uint32_t location_id = row[index["loc_id_end"]];
+        uint32_t activity    = row[index["activity_end"]];
+
+        auto it_location              = locations.find(location_id);
+        mio::abm::LocationId location = it_location->second;
+        if (it_location == locations.end()) {
+            location = world.add_location(get_location_type(activity), 0); // TODO: adjust LocationType
+            locations.insert({location_id, location});
+        }
+        auto it_person = persons.find(person_id);
+        if (it_person == persons.end()) {
+            auto& person =
+                world.add_person(location, mio::abm::InfectionState::Susceptible, static_cast<mio::abm::AgeGroup>(age));
+            persons.insert({person_id, person});
+            it_person = persons.find(person_id);
+        }
+        it_person->second.set_assigned_location(location);
```

Comment:
Only set this when the location type is not home.

Reply:
Changed.